backend/src/router.rs

```diff
@@ -90,39 +92,82 @@ where
     }
 
     /// Add API Route to the `Router`
-    ///
     #[must_use]
-    pub fn api_route<H, T>(mut self, path: &str, method: &Method, handler: H) -> Self
+    #[cfg(not(all(feature = "swagger", debug_assertions)))]
+    pub fn api_route<H, T>(self, path: &str, method: &Method, handler: H) -> Self
     where
         H: Handler<T, S, B>,
         T: 'static,
         S: Clone + Send + Sync + 'static,
         B: HttpBody + Send + 'static,
     {
-        #[cfg(all(feature = "swagger", debug_assertions))]
-        match try_convert_path_item_type_from_method(method)
-            .map(|path_item_type| check_api::<_, _, _, H, Version, Doc>(path, &path_item_type))
-        {
-            Ok(Ok(())) => (),
-            Ok(Err(err)) | Err(err) => {
+        self.apply_handler::<H, T>(path, method, handler)
+    }
+
+    fn apply_handler<H, T>(mut self, path: &str, method: &Method, handler: H) -> Self
+    where
+        H: Handler<T, S, B>,
+        T: 'static,
+        S: Clone + Send + Sync + 'static,
+        B: HttpBody + Send + 'static,
+    {
+        match try_convert_method_filter_from_method(method) {
+            Ok(method) => self.inner = self.inner.route(path, on(method, handler)),
+            Err(err) => {
                 if let Some(errors) = &mut self.api_errors {
                     errors.extend_one(err);
                 } else {
                     self.api_errors = Some(err);
                 }
             }
-        }
+        };
 
-        match try_convert_method_filter_from_method(method) {
-            Ok(method) => self.inner = self.inner.route(path, on(method, handler)),
-            Err(err) => {
+        self
+    }
+}
+
+#[cfg(all(feature = "swagger", debug_assertions))]
```

Comment:
The separation of implementations occurs at different levels. Here the `struct` level is used, and above - the function level. It is better to unify places, where `cfg` is applied. It seems that it is better to make a separation at the function level

Reply:
The only difference seems to be the call to `check_api`, I suggest to `cfg` only the call and the function itself